src/libraries/Microsoft.Extensions.Options/gen/Emitter.cs

```diff
@@ -209,10 +604,17 @@ private void GenModelSelfValidationIfNecessary(ValidatedModel modelToValidate)
             OutLn($"/// <returns>Validation result.</returns>");
             OutGeneratedCodeAttribute();
 
+            // We disable the warning on `new ValidationContext(object)` usage as we use it in a safe way that not require executing the reflection code.
+            // This is done by initializing the DisplayName in the context which is the part trigger reflection if it is not initialized.
+            OutLn($"#if NET5_0_OR_GREATER // UnconditionalSuppressMessage is supported on .NET 5.0 and later");
```

Comment:
Is it not possible to read the target framework from the `Compilation` object and emit the suppression conditionally? I would assume this should work even in multi-targeted projects since the source generator runs independently for every target.

Reply:
I am thinking of the library author scenario who produce library which is be used later in AOT app. what do you think about that?

Reply:
I applied the suggestion